### PR TITLE
[react, wallets] Add Frame wallet

### PR DESCRIPTION
**Pull request by goosewobbler**
Adding a new connector for [Frame](https://frame.sh)

### CODE REVIEW

**Comment by joaquim-verges** [comment]
@goosewobbler excited to get this merged - let us know what you need any help, looks like you got it tho. 

Also curious to hear what you think of the API, feedback is appreciated!

**Comment by goosewobbler** [comment]
@joaquim-verges I wasn't able to test it locally so far, I've been following the instructions but getting compilation errors with missing FrameWallet export in `@thirdweb-dev/wallets`.  The export is definitely there so it must be a local issue...

**Comment by MananTank** [comment]
@goosewobbler I tried to replicate the issue, but it seems that import is working. If it's not working for you, I suggest deleting the entire `node_modules/@thirdweb-dev` folder and yalc-ing all the packages


```bash
rm -rf node_modules/@thirdweb-dev

# delete cache - example: vite
rm -rf node_modules/.vite

yalc add @thirdweb-dev/chains @thirdweb-dev/contracts-js @thirdweb-dev/generated-abis @thirdweb-dev/react @thirdweb-dev/react-core @thirdweb-dev/sdk @thirdweb-dev/storage @thirdweb-dev/wallets
```

**Comment by MananTank** [comment]
Hi @goosewobbler - Is this PR ready for testing?

**Comment by goosewobbler** [comment]
We're waiting on a new page for diagnosing connection issues on our docs site so we can link to it in the connection UI.  Will mark as ready for review though, can always follow up.

**Comment by MananTank** [comment]
Hey @goosewobbler, I am not able to connect the frame wallet. Can you look into the issue here? 

https://github.com/thirdweb-dev/js/assets/22043396/1a4d325d-151e-4cd0-a0dd-c2325e4f4e3a

**Comment by MananTank** [comment]
> @MananTank What chain have you set your localhost dapp to? There's a lot of missing chain errors there, I would guess that you are trying to connect to Polygon without it being enabled in Frame.

I tried, ethereum and goerli - both didn't work. I was able to connect once - then I disconnected the localhost from dapp connections and I was never able to connect again

**Comment by goosewobbler** [comment]
> I tried, ethereum and goerli - both didn't work. I was able to connect once - then I disconnected the localhost from dapp connections and I was never able to connect again

Frame doesn't handle disconnection like this, did you turn off the permission for the dapp?

**Comment by MananTank** [comment]
Hi @goosewobbler - Sorry for the late response, I've followed your steps and was able to get it working 👍

File: packages/react/src/wallet/wallets/frame/frameWallet.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    iconURL:
+      "https://frame.nyc3.digitaloceanspaces.com/bundle/home/favicon.8f0e1342.png",
+    urls: {
+      chrome:
+        "https://chrome.google.com/webstore/detail/frame-companion/ldcoohedfbjoobcadoglnnmmfbdlmmhf",
+      firefox: "https://addons.mozilla.org/en-US/firefox/addon/frame-extension",
+    },
+  },
+  create(options: WalletOptions) {
+    const walletStorage = createAsyncLocalStorage(walletIds.frame);
```
Comment (MananTank) [inline packages/react/src/wallet/wallets/frame/frameWallet.tsx:23]: This is not required - it is done automatically

Comment (MananTank) [inline packages/react/src/wallet/wallets/frame/frameWallet.tsx:23]: creating `walletStorage` here is not required - `createAsyncLocalStorage` is used by default

**Comment by MananTank** [comment]
@goosewobbler - Everything looks. Just two outstanding issues 

* the `<GetStartedScreen/>` is not being used in frame wallet connect UI
* if the frame wallet app is not running, the connection fails - We should display a UI to let the user know what to do in that case

**Comment by goosewobbler** [comment]
@MananTank I addressed these two issues, should be good to go now.

**Comment by goosewobbler** [comment]
@MananTank Not sure what is going on there, it connects for me without the extension.  Do you have any errors in the dev console?

**Comment by MananTank** [comment]
@goosewobbler This is the error I'm getting

image

File: .changeset/ten-donuts-pretend.md
Comment (iketw) [inline .changeset/ten-donuts-pretend.md:1]: @goosewobbler , tested this out and it works great for me. I added this changeset but feel free to add more info in the file if you want to. This is one of the places where devs will read about Frame and how to set it up.

Once you give the green light we'll commit this one.

Thanks for submitting this PR!!

Comment (goosewobbler) [reply .changeset/ten-donuts-pretend.md:1]: Just a minor wording update, good to go I think!

**Comment by iketw** [comment]
@goosewobbler, sorry for not mentioning this before but we encourage you to also add a brief description of how to create the Frame wallet in our documentation/portal. This is where devs go first to check which wallets are available.

Please, see an example here: https://github.com/thirdweb-dev/docs/blob/main/docs/onboarding/14%20Wallet/1%20Wallets/MetaMask.mdx

**Pull request merged**